content/en/docs/components/pipelines/author-a-pipeline/inputs-outputs-data-passing.md

```diff
@@ -0,0 +1,349 @@
+
+|        | Parameter        | Artifact
+| ------ | ---------------- | --------------- |
+| Input  | Input Parameter  | Input Artifact  |
+| Output | Output Parameter | Output Artifact |
+
+Throughout the remainder of this section, we will use the following example dataset creation pipeline to understand the behavior and usage of input and output parameters and artifacts:
+
+```python
+from kfp import dsl
+from kfp.dsl import Input, Output, Dataset
+
+
+@dsl.container_component
+def create_dataset(
+    initial_text: str,
+    output_dataset: Output[Dataset],
+):
+    """Create a dataset containing the string `initial_text`.""" 
+    return dsl.ContainerSpec(
+        image='alpine',
+        command=['sh', '-c', 'mkdir --parents $(dirname "$1") && echo "$0" > "$1"',],
+        args=[initial_text, output_dataset.path])
+
+
+@dsl.component
+def augment_dataset(
+    existing_dataset: Input[Dataset],
+    resulting_dataset: Output[Dataset],
+    text: str,
+    num: int = 10,
+) -> int:
+    """Append `text` `num` times to an existing dataset, then write it as a new dataset."""
+    additional_data = ' '.join(text for _ in range(num))
+
+    with open(existing_dataset.path, 'r') as f:
+        existing_dataset_text = f.read()
+
+    resulting_dataset_text = existing_dataset_text + ' ' + additional_data
+
+    with open(resulting_dataset.path, 'w') as f:
+        f.write(resulting_dataset_text)
+
+    return len(resulting_dataset_text)
+
+
+@dsl.pipeline()
+def my_pipeline(initial_text: str = 'initial dataset text'):
+    create_task = create_dataset(initial_text=initial_text)
+    augment_dataset(
+        existing_dataset=create_task.outputs['output_dataset'],
+        text='additional text')
+```
+
+This pipeline uses a container component `create_dataset` to construct an initial `Dataset` artifact containing `initial_text`. Then, the downstream lightweight Python component `augment_dataset` appends `text` repeated `num` times to the dataset and saves it as a new dataset.
+
+## Inputs
+Component inputs are specified by the component function's signature. This applies for all authoring approaches: [lightweight Python components][lightweight-python-component], [containerized Python components][containerized-python-component], and [container components][container-component].
+
+Ultimately, each authoring style creates a component definitied by an `image`, `command`, and `args`. When you use an input, it is represented as a placeholder in the `command` or `args` and is interpolated at component runtime.
+
+There is one additional type of input, `PipelineTaskFinalStatus`, which allows access to the metadata of one task from within another. This input is a special case, as it is neither a parameter nor an artifact. Use of this input is covered in [Authoring: Pipelines][pipelines].
```

Comment:
Technically, `PipelineTaskFinalStatus` is a struct typed parameter with system provided value. And it's only usable in an exit handler scenario. 
But feel free to refer to other doc for more details on this.